bundles/AdminBundle/src/Controller/Admin/SettingsController.php

```diff
@@ -528,6 +529,10 @@ public function setSystemAction(
         $settingsYaml = Yaml::dump($settings, 5);
         $configFile = Config::locateConfigFile('system.yaml');
         File::put($configFile, $settingsYaml);
+        if ($existingValues) {
+            $saveSettingsEvent = new GenericEvent(null, [$existingValues, $values]);
```

Comment:
```suggestion
            $saveSettingsEvent = new GenericEvent(null, [
                'settings' => $settings, 
                'existingValues' => $existingValues, 
                'values' => $values,
            ]);
```
We should used array keys here and also pass the final settings that get persisted. 

I also think it would make sense to do this before the settings get persisted to a YAML file, because then we could offer the possibility to change values before. 
Just as we do it e.g. here: 
https://github.com/pimcore/pimcore/blob/64e31dba49c1a5c8ee584f2e30efc6834ce5eb16/bundles/AdminBundle/src/Controller/Admin/Asset/AssetController.php#L239
by using `$saveSettingsEvent->getArgument('settings')`

Reply:
Thank you for the review. Code should be adapted properly now.